src/hooks/useClusterTemplates.ts

```diff
@@ -18,3 +18,20 @@ export const useClusterTemplate = (name: string): [ClusterTemplate, boolean, unk
     groupVersionKind: clusterTemplateGVK,
     name: name,
   });
+
+export const useClusterTemplatesFromRepo = (
+  repoUrl?: string,
+): [ClusterTemplate[], boolean, unknown] => {
+  const [templates, loaded, error] = useClusterTemplates();
+  if (!repoUrl || !loaded || error) {
+    return [[], loaded, error];
+  }
+  const repoTemplates = repoUrl
+    ? templates.filter(
+        (template) =>
+          template.spec.clusterDefinition.source.repoURL === repoUrl ||
+          template.spec.clusterSetup?.find((spec) => spec.spec.source.repoURL === repoUrl),
+      )
+    : [];
+  return [repoTemplates, loaded, error];
```

Comment:
no need for the `if` 
```suggestion
  const repoTemplates = React.useMemo(() => repoUrl
    ? templates.filter(
        (template) =>
          template.spec.clusterDefinition.source.repoURL === repoUrl ||
          template.spec.clusterSetup?.find((spec) => spec.spec.source.repoURL === repoUrl),
      )
    : [], [repoUrl, templates]);
  return [repoTemplates, loaded, error];
```